### DISCUSSION TITLE
Automatically "contain" images that have transparency

### DISCUSSION THREAD

**Discussion by LasseRosenow** [directus/directus#10964] [Feature Requests]
When rendering images in previews, directus uses `cover` so that the preview box is always completely filled. This is very good for images that have a background, but for images with a transparent background this is very much unnecessary and usually creates a bad visual:

![image](https://user-images.githubusercontent.com/10547444/148651216-14c9d8a0-ee61-4933-946d-1f337934580a.png)

I think if images with transparent background are rendered with `contain` by default, this would improve the default UI a lot :)

**Comment by oreilles** [comment]
The problem isn't that those images have pixel with alpha values, but that they display text which you'd like to see all rendered. Unfortunately that's not something we can detect, and that's not something everyone would agree either.

This cannot be considered a bug so I'm going to close it - but feel free to open a feature request in the Discussions section. 👍

**Comment by benhaynes** [comment]
I do believe that we have options for cover versus contain on the File Library and each Field Interface... so these options are there... just not on an automatic case-by-case basis. Please correct me if I'm wrong!

**Comment by LasseRosenow** [comment]
I did not mean it in a way, that the alpha value is the problem. And no it is not the text that I want to see. I just think that in general all images that have alpha values always look better if they are rendered with `contain` instead of `cover`.

The only reason to render something with `cover` is so that it fits the box and there are no weird paddings and stuff, but if an image has alpha values, it does not fill the box no matter if it is scaled up or not so they would look better if they are scaled down instead of scaled up

**Comment by rijkvanzanten** [comment]
I don't think there's a way for us to default that behavior in an unopinionated way 🤔 What @benhaynes said is correct; there's a dedicated toggle to control the cover vs contain that I'd recommend using in this case:

<img width="533" alt="CleanShot 2022-01-10 at 13 33 22@2x" src="https://user-images.githubusercontent.com/9141017/148819930-82cbc622-50bb-49a4-8aa0-9974da27b681.png">

**Comment by LasseRosenow** [comment]
> I do believe that we have options for cover versus contain on the File Library and each Field Interface... so these options are there... just not on an automatic case-by-case basis. Please correct me if I'm wrong!

This is true, but it only works on a per image field basis. Not for the gallery for example. I could change it for the whole gallery, but then all images with backgrounds would look bad.

This idea of using `cover` for images with full colored backgrounds and `contain` for images with transparent backgrounds is just my proposal on how to have better defaults.

My Idea would also be to update the interface so that there would be 3 options:
1. `auto`: images with alpha use `contain` images with full colored backgrounds use `cover`
2.  `cover`: image will be forced to be displayed as `cover`
3.  `contain`: image will be forced to be displayed as `contain`

**Reply by rijkvanzanten** [reply]
Makes sense! Moved this to a feature request discussion to track it as such 🙂 

An interesting side effect of that would be that it won't actually work for the image in the middle of your original screenshot, as that seemingly doesn't have any transparency 🤔

**Reply by LasseRosenow** [reply]
Thank you :)

This is correct, that image is a huge rarity though :D, but there will never be a perfect solution that works in every case if it is not ai based :D 

The goal is just to get it as good as it gets :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
